nuget/lib/dependabot/nuget/file_updater/project_file_declaration_finder.rb

```diff
@@ -95,6 +111,80 @@ def declaring_file
 
           raise "No file found with name #{filename}!"
         end
+
+        def sdk_import_strings
+          sdk_strings(SDK_IMPORT_REGEX, "Import", "Sdk", "Version")
+        end
+
+        # rubocop:disable Metrics/PerceivedComplexity
+        def sdk_project_strings
+          dep_name = dependency_name&.downcase
+          dep_version = declaring_requirement.fetch(:requirement)
+          strings = []
+          declaring_file.content.scan(SDK_PROJECT_REGEX).each do |string|
+            xml = string
+            xml += "</Project>" unless string.end_with?("/>")
+            node = Nokogiri::XML(xml)
+            node.remove_namespaces!
+            element = node.at_xpath("/Project")
+            next unless element
+
+            sdk_references = element.attribute("Sdk")&.value ||
+                             element.attribute("sdk")&.value
+            sdk_references = sdk_references&.strip
+            next unless sdk_references&.include?("/")
+
+            sdk_references.split(";").each do |sdk_reference|
+              parts = sdk_reference.split("/")
+              next unless parts.length == 2
+              next unless parts[0]&.downcase == dep_name
+              next unless parts[1] == dep_version
+
+              strings << sdk_reference
+            end
+          end
+          # Drop duplicates, to avoid potential issues with repeated replacements (e.g. upgrading Foo/1.0.0-beta to
+          # Foo/1.0.0-beta.2). However, that turns the empty array into nil.
+          strings.uniq! || []
+        end
+        # rubocop:enable Metrics/PerceivedComplexity
+
+        def sdk_sdk_strings
+          sdk_strings(SDK_SDK_REGEX, "Sdk", "Name", "Version")
+        end
+
+        # rubocop:disable Metrics/CyclomaticComplexity
+        # rubocop:disable Metrics/PerceivedComplexity
```

Comment:
I think there might be some room to extract certain parts of this into separate methods, the linter things the complexity of this method is too high and I kind of agree that it's a little hard to follow.

I do also understand that you've submitted this PR a really long time ago and you might not have an appetite to dig back in and refactor things, but it would be appreciated if we can clean this up a little and reduce the complexity of the method a little.

Reply:
Done